sphinx/ext/autosummary/__init__.py

```diff
@@ -512,8 +513,7 @@ def _import_by_name(name):
         modname = '.'.join(name_parts[:-1])
         if modname:
             try:
-                __import__(modname)
-                mod = sys.modules[modname]
+                mod = import_module(modname)
```

Comment:
I think this is not equivalent:
```__import__('sphinx.ext')```
returns module `sphinx`, not `sphinx.ext`.

Reply:
Fix in #4234

Reply:
Absolutely right.